### PR TITLE
Move to SieveCache

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/gallery/PageLoader.kt [hunk 1]
```diff
@@ -15,14 +15,14 @@ private const val MIN_CACHE_SIZE = 128 * 1024 * 1024
 
 abstract class PageLoader {
     private val cache by lazy {
-        lruCache<Int, Image>(
+        SieveCache<Int, Image>(
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/gallery/PageLoader.kt:18]: The androidx implementation of SieveCache is not thread-safe, so we need to make sure there's no concurrent modification.

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/gallery/PageLoader.kt]: Done

**Review by FooIbar — APPROVED** [review]
Ideally we should use a RW lock to allow concurrent read-only access.

**Comment by revonateB0T** [comment]
> Ideally we should use a RW lock to allow concurrent read-only access.

I will do that with next MR

**Pull request merged**